File: solutions/cpp/rna-transcription/rna_transcription.cpp
#include "rna_transcription.h"
#include <map>

namespace rna_transcription
{
    using namespace std;

    map<char, char> dna_to_rna = {
        {'G', 'C'},
        {'C', 'G'},
        {'T', 'A'},
        {'A', 'U'},
    };

    char to_rna(char nucleotide)
    {
        return dna_to_rna[nucleotide];
    }

    string to_rna(string dna)
    {
        size_t length = dna.size();
        string rna(length, '?');

        for (size_t i = 0; i < length; i++)
        {
            rna[i] = to_rna(dna[i]);
        }
        return rna;
    };
} // namespace rna_transcription

File: solutions/go/rna-transcription/rna_transcription.go
package strand

// mapping of each DNA nucleotide to its RNA complement
var dnaToRna = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

// ToRNA transcribes a string representing DNA nucleotides to RNA.
func ToRNA(dna string) string {
	rna := make([]rune, len(dna))

	for i, nucleotide := range dna {
		rna[i] = dnaToRna[nucleotide]
	}
	return string(rna)
}
